impl/src/planning/crate_catalog.rs

```diff
@@ -140,6 +142,8 @@ impl CrateCatalogEntry {
           format!("{}/vendor/{}", settings.workspace_path, &self.package_ident)
         }
       },
+      // Settings should always have `genmode` set to one of the above fields
+      GenMode::Unspecified => unreachable!(),
```

Comment:
roughly the same here -- First preference is Result<String>, second preference is explicit panic, final and least preferred is to attach a better explanation for why this is "unreachable".